packages/components/addon/components/hds/form/character-count/index.hbs

```diff
@@ -0,0 +1,29 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<Hds::Text::Body
+  @tag="div"
+  @size="100"
+  class={{this.classNames}}
+  id={{this.id}}
+  {{did-insert this.onInsert}}
+  {{will-destroy this.willDestroyNode}}
+  ...attributes
+  aria-live="polite"
+>
+  {{#if (has-block)}}
+    {{yield
+      (hash
+        minLength=this.minLength
+        maxLength=this.maxLength
+        currentLength=this.currentLength
+        remainingLength=this.remainingLength
+      )
+    }}
+  {{else}}
+    {{this.currentLength}}
```

Comment:
I would start to add the logic for the different use cases here, even if the copy is not finalized yet, so we can start to see them in action (see comment below about adding a "live/working" use case in the showcase)